### ISSUE TITLE
container.logs seems to no stream logs when the container is started without the tty option 

### ISSUE THREAD

**Issue by gpolaert** [docker/docker-py#1394]
I'm not sure if it's an issue.

I try to stream container logs using the docker-py client.
I noticed I have no logs when my container is started without `-t`.

```
logs = container.logs(stream=True, follow=True, stdout=True, stderr=False, tail=0)
````

Even if I can show them using this command `docker logs -f <container>`

**Comment by gpolaert** [comment]
Could someone help me to understand?

**Comment by TomasTomecek** [comment]
Did you try to tinker switches? I would definitely start with:

```
logs = container.logs(stream=True, follow=True, stdout=True, stderr=True, tail='all')
```

It also depends what kind of application you are running in the container.


Also bear in mind that 

```
$ docker logs -f <container>
```

is not an equivalent of

```
logs = container.logs(stream=True, follow=True, stdout=True, stderr=False, tail=0)
```

**Comment by danqing** [comment]
Run into a similar problem. @TomasTomecek can you explain the difference between the two, and also what's the difference between `stream` and `follow`? What happens if you specify one or both? Why does `stream` not `follow`? Does `follow` automatically `stream`? What is returned when both are specified?

**Comment by TomasTomecek** [comment]
@danqing since my comment is 6 months old, I forgot everything; I would need to check sources to answer, so please see the sources for more info

**Comment by danqing** [comment]
Yeah the source is pretty confusing too :( I've hacked a solution, but it looks like the logging part of this library still needs some significant improvements to work reliably.

**Comment by dperetti** [comment]
Facing a similar issue with `detach=False`.
with `detach=False, tty=false`(the default): STDOUT ok, but no STDERR.
with `detach=False, tty=true`: STDERR ok, but no STDOUT! 🤔

**Comment by kode54** [comment]
8 years later, this issue is affecting TrueNAS Scale, and they referred me to this issue, saying they "depend on upstream to fix it for them".

**Comment by TannerLivingston** [comment]
This is finally resolved for me on version ElectricEel-24.10.2.2.

**Comment by ArcticLampyrid** [comment]
@TomasTomecek Could you please tell me how do you solve it?
